.github/workflows/ci.yml

```
@@ -0,0 +1,24 @@
+# https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: ci
+on:
+  push:
+  #  branches: [main]
+  pull_request:
+    branches: [main]
+jobs:
```

```suggestion
permissions:
  contents: read
jobs:
```

Good to explicitly set permissions here.